rest-api/rest-api.php

```diff
@@ -0,0 +1,436 @@
+        $res = WP_Query_Push::instance()->process_task($connection_id, $query);
+        $response_data = $res['response_data'];
+        $errors = $res['errors'];
+        $status = $res['status'];
+        /*
+        if ( empty ( $response_data ) ) {
+            $response_data = $errors;
+        }
+        */
+        return wp_send_json( [], 200 );
+    }
+
+    public function handle_post_interval( WP_REST_Request $request ) {
+        // TODO: parse and validate request data
+        $key = 'every_five_seconds';
+        $interval = 5;
+        $display = 'Every Five Seconds';
+        $schedules = wp_get_schedules();
+        $schedules[$key] = array(
+            'interval' => $interval,
+            'display'  => $display
+        );
+        return wp_send_json( array( 'id' => $key ), 200 );
+    }
+
+    public function parse_request_query( WP_REST_Request $request ) {
+        $body = $request->get_body();
+        $data = json_decode($body, true);
+        $query = $data['statement'];
+        return $query;
+    }
+
+    public function handle_post_save_query( WP_REST_Request $request ) {
+        try {
+            $query = $this->parse_request_query($request);
+            if ( empty( $query ) ) {
+                return wp_send_json( [ 'error' => 'Bad Request' ], 400 );
+            }
+
+            global $wpdb;
+            $table_name = $wpdb->prefix . WP_Query_Push::instance()->TABLE_NAME_QUERIES;
+            $wpdb->insert(
+                $table_name,
+                [ 'query' => $query ],
+                [ "%s" ] );
+            return wp_send_json( [ 'id' => $wpdb->insert_id ], 200 );
+        } catch ( Exception $e ) {
+            return wp_send_json( [ 'error' => 'Server Error' ], 500 );
+        }
+    }
+
+    public function handle_query( WP_REST_Request $request ) {
+        try {
+            $query = $this->parse_request_query( $request );
+            if ( empty( $query ) ) {
+                return wp_send_json( [ 'error' => 'Bad Request' ], 400 );
+            }
+            return $this->run_query_json( $query );
+        } catch ( Exception $e ) {
+            return wp_send_json( [ 'error' => 'Server Error' ], 500 );
+        }
+    }
+
+    public function get_api_key( WP_REST_Request $request ) {
+        $api_key = get_option( 'wpquerypush_api_key' );
+        if ( $api_key ) {
+            return $api_key;
+        }
+        return false;
+    }
+
+    private function generate_api_key() {
+        $random_bytes = random_bytes(32);
+        $api_key = bin2hex($random_bytes);
+        return $api_key;
+    }
+
+    public function create_new_api_key( WP_REST_Request $request ) {
+        $new_api_key = self::generate_api_key();
+        $existing_api_keys = get_option( 'wpquerypush_api_keys' );
+        if ( empty( $existing_api_keys ) ) {
+            $existing_api_keys = [];
+        }
+        array_push( $existing_api_keys, $new_api_key );
+        set_option( 'wpquerypush_api_keys', );
```

Comment:
@prykon is the comma following the `'wpquerypush_api_keys'` string not causing an error?

Reply:
Updated.